### ISSUE TITLE
New rule: `use-new-qunit-tests`

### ISSUE THREAD

**Issue by jbandura** [ember-cli/eslint-plugin-ember#188]
## Proposal
(_The name could probably use some tweaking (man, naming is hard_ 😓 )

Goal of this rule is to enforce usage of the new-style qunit tests: http://rwjblue.com/2017/10/23/ember-qunit-simplication/

It should highlight places where we're using old style qunit tests, using `moduleForComponent` or `moduleFor`. This could come especially handy when converting to new test syntax in bigger teams (where for some reason using codemod is out of the question) and also to enforce uniform usage (since the old style is still working).

## Example (taken from RFC):
Component: 
```js
//=========
// bad
//=========

import { moduleForComponent, test } from 'ember-qunit';
//...

moduleForComponent('x-foo', {
  integration: true
});

//...

//==========
// good
//==========

import { module, test } from 'qunit';
import { setupRenderingTest } from 'ember-qunit';
//...

module('x-foo', function(hooks) {
  setupRenderingTest(hooks);
  //..
});
```
Service/Controller/Route:
```js
//==========
// bad
//==========

import { moduleFor, test } from 'ember-qunit';

moduleFor('service:flash', 'Unit | Service | Flash', {
  unit: true
});

//...

//==========
// good
//==========

import { module, test } from 'qunit';
import { setupTest } from 'ember-qunit';

module('Unit | Service | Flash', function(hooks) {
  setupTest(hooks);

  //...
});
```

**Comment by tmquinn** [comment]
I am going to work on this next week

**Comment by rwjblue** [comment]
Awesome!

**Comment by rwjblue** [comment]
This landed and is included in 5.3.0, will be in the recommended set with 6.0.0.

**Comment by rwjblue** [comment]
Whoops! I was wrong, sorry about that. It is in https://github.com/ember-cli/eslint-plugin-ember/pull/281.

**Comment by Mifrill** [comment]
any updates one this one?

**Comment by bmish** [comment]
@Mifrill: We do have an existing rule [no-test-module-for](https://github.com/ember-cli/eslint-plugin-ember/blob/master/docs/rules/no-test-module-for.md). I think that addresses most of what this issue was talking about? 

Although it looks like there was a PR for a new rule `use-nested-qunit-syntax` (https://github.com/ember-cli/eslint-plugin-ember/pull/281 ) that got lost.

**Comment by Mifrill** [comment]
> We do have an existing rule no-test-module-for.

@bmish yeah, that's what I need, thank you!
